tools/jit/gen_jit_dispatch.py

```diff
@@ -274,16 +270,27 @@ def declkey(decl):
                 ])
                 decl['has_tensor_options'] = True
 
-    jit_decls = sort_decls(jit_decls)
-    for decl in jit_decls:
-        ops.append(OPERATOR.substitute(signature=signature(decl),
-                                       op=emit_decl_variant(decl)))
-
-    # Sort the generated snippets to ensure that the generation is deterministic
-    env = {
-        'constructors': ops,
-    }
-    write(out, 'register_aten_ops.cpp', REGISTER_ATEN_OPS_CPP, env)
+    # Group and sort the generated snippets to ensure that the
+    # generation is deterministic
+    jit_decl_groups = sort_decls(jit_decls)
+
+    # NOTE: see the comment at the top of the register_aten_ops.cpp
+    # template regarding sharding of the generated files.
+    num_shards = 3
+    shards = [[] for _ in range(num_shards)]
+
+    # ops are assigned arbitrarily but stably to a file based on hash
+    for group in jit_decl_groups:
+        x = sum(ord(c) for c in group[0]['name']) % num_shards
+        for decl in group:
+            shards[x].append(OPERATOR.substitute(signature=signature(decl),
+                                                 op=emit_decl_variant(decl)))
+
+    for i, shard in enumerate(shards):
+        env = {
+            'constructors': shard,
+        }
+        write(out, 'register_aten_ops-%d.cpp' % i, REGISTER_ATEN_OPS_CPP, env)
```

Comment:
This is not portable to Python 3. Use 'register_aten_ops-{}.cpp'.format(i) instead